design/secrets_provider_milestone1_design.md

```diff
@@ -0,0 +1,497 @@
+
+As a result, one can define the host representing the Secrets Provider in Conjur policy using only **namespace** and/or **service account** granularities.
+
+###### How the drawbacks can be handled?
+
+To handle the missing **Job** granularity there are 2 options:
+
+|      | Solution                                                     | Pros                                                         | Cons                                                         | Effort Estimation |
+| ---- | ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------------------------------------------ | ----------------- |
+| 1    | Keep using only **namespace** and/or **service account** granularities. | - Since Secrets Provider runs as a separate entity, we suggest  creating service account dedicated for Secrets Provider.<br />Then, **service account** granularity serves exact match and **Job** granularity is redundant<br />- No code changes | - Service account might be reused, which allows other apps to authenticate using Secrets Provider host | Free              |
+| 2    | Add support for **Job** granularity in Conjur                | - Allows specific identification of the Secrets Provider **Job**, which enhances security | - Costly; requires code changes, verify compatibility, tests, docs<br />- Redundant if **service account** serves only the Secrets Provider | 10 days           |
+
+*Decision*: Solution #1, use **namespace** and **service account** to define the Secrets Provider host in Conjur policy.
+Recommend to the customer to create a dedicated service account for Secrets Provider.
+
+#### Customer Experience
+
+1. Configure K8s Secrets with ***conjur-map\*** metadata defined 
+
+   - *Optional:* add label to K8s Secret
+
+2. Create Service Account for Secrets Provider 
+
+3. Create Role for the Service Accounts with **get**/**update** and also **list** privileges on K8s Secrets resources
+
+4. Create RoleBinding and bind the Service Accounts to the Role from previous step
+
+5. Create/Deploy Secrets Providers as Jobs in deployment manifest(s)
+   - If label filtering is defined, add "K8S_SECRET_LABEL" in the Secrets Provider manifest(s)
+
+The Secrets Provider host will resemble the following:
+
+```yaml
+- !policy
+  id: secrets-accessors
+  body:
+  - !host
+    id: secrets-provider-1
+    annotations:
+      authn-k8s/namespace: prod-namespace
+      authn-k8s/service-account: prod-sa
+      authn-k8s/authentication-container-name: cyberark-secrets-provider
+
+  - !group 
+
+  - !permit
+    resources: !variable super_secret
+    role: !group secrets-accessors
+    privileges: [ read, execute ]
+ 
+  - !grant
+    role: !group secrets-accessors
+    members:
+      - !host conjur/authn-k8s/secrets-provider-1
+```
+
+The Secrets Provider manifest will be defined as the following:
+
+```yaml
+apiVersion: batch/v1
+kind: Job
+metadata:
```

Comment:
where is the namespace (`prod-namespace`) is being declared?

Reply:
added (: